reference-implementation/lib/writable-stream.js

```diff
@@ -173,11 +176,26 @@ function WritableStreamFinishClose(stream) {
   defaultWriterClosedPromiseResolve(stream._writer);
 }
 
-function WritableStreamFulfillWriteRequest(stream) {
-  assert(stream._writeRequests.length > 0);
+function WritableStreamRejectUnresolvedPromises(stream) {
```

Comment:
I wonder if there's a better name for this that indicates clearly that it will reject write requests + pending close requests + writer closed promise, but not pending write requests.

Also, can it maybe assert that there is no pending write request?